.github/actions/check-pr-format/action.yml

```diff
@@ -0,0 +1,38 @@
+name: PR Title Check
+on:
+  pull_request:
+    types:
+      - opened
+      - edited
+      - synchronize
+
+jobs:
+  title:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check PR title
+        run: |
+          const title = context.payload.pull_request.title;
+          const regex = /\[(Add|Remove|Update|Fix)\]\s*.+/;
+          const regexNoBrackets = /(Add|Remove|Update|Fix)\s*.+/i;
+          const regexNoSpace = /\[(Add|Remove|Update|Fix)\].+/i;
+          const regexLowerCase = /\[(add|remove|update|fix)\]\s*.+/i;
+          const issue_number = context.payload.pull_request.number;
+          const owner = context.repo.owner;
+          const repo = context.repo.repo;
+          let body = '';
```

Comment:
This can be defined within the first if. We don't use it outside of the if cases below.

Reply:
I'm not sure what you mean, if it's defined in the first if then it's not available in the other cases. Maybe I'm misunderstanding your comment?

Reply:
Apologies for not being very clear. I couldn't see body being used outside of the if case. It can be moved inside the first if I think. Like this:
```
if (!regex.test(title)) {
            let body = '';
            if (regexNoSpace.test(title)) {
              body = 'Your PR title is missing a space after the keyword. Please update it to: [Add | Remove | Update | Fix] <Title>. For example: Remove Old feature';
            } else if (regexLowerCase.test(title)) {
              body = 'The keyword in your PR title is not capitalized. Please update it to: [Add | Remove | Update | Fix] <Title>. For example: Update Existing feature';
            } else {
              body = 'Your PR does not have a [Add | Remove | Update | Fix] <Title>. Please add one to help streamline our changelog. For example: Fix Bug in feature';
            }
            github.issues.createComment({ issue_number, owner, repo, body });
}
```

Reply:
That's true, though I have a habit of declaring all variables outside of conditionals as it reduces cognitive load to declare everything you're going to work with ahead of the logic.